### ISSUE TITLE
Even after a template is removed from catalog , it is listed in the catalog view.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#2431]
Server version - V0.43.0-rc2

"Jenkins-CI" was removed ~ 3 hours back form rancher/rancher-catalog.

But it is still listed in the catalog view.
<img width="1101" alt="screen shot 2015-10-24 at 3 47 01 pm" src="https://cloud.githubusercontent.com/assets/4266958/10713188/5b1445f4-7a68-11e5-8cde-5ff29832ee28.png">

DATA folder in my rancher-server container does not have this template.

but <rancher-server>::8080/v1-catalog/templates returns jenkins-ci.

```
{"type":"collection","resourceType":"template","data":[{"type":"template","links":{"self":"http://104.197.155.38:8080/v1-catalog/templates/jenkins-ci"},"name":"Jenkins","uuid":"","category":"Continuous Integration","description":"Jenkins CI server (LTS)\n","version":"1.625.1-rancher1","iconLink":"http://104.197.155.38:8080/v1-catalog/images/jenkins-ci/catalogIcon-jenkins-ci.svg","versionLinks":{"1.625.1-rancher1":"http://104.197.155.38:8080/v1-catalog/templates/jenkins-ci/0"},"dockerCompose":"","rancherCompose":"","questions":null,"path":"jenkins-ci"},{"type":"template","links":{"self":"http://104.197.155.38:8080/v1-catalog/templates/elasticsearch"},"name":"Elasticsearch","uuid":"","category":"ELK","description":"Elasticsearch, you know for search!\n","version":"1.7.3-rancher1","iconLink":"http://104.197.155.38:8080/v1-catalog/images/elasticsearch/catalogIcon-elasticsearch.svg","versionLinks":{"1.7.1-rancher1":"http://104.197.155.38:8080/v1-catalog/templates/elasticsearch/0","1.7.3-rancher1":"http://104.197.155.38:8080/v1-catalog/templates/elasticsearch/1"},"dockerCompose":"","rancherCompose":"","questions":null,"path":"elasticsearch"},{"type":"template","links":{"self":"http://104.197.155.38:8080/v1-catalog/templates/kibana"},"name":"Kibana 4","uuid":"","category":"ELK","description":"Visualiztion dashboard","version":"4.1.1-rancher1","iconLink":"http://104.197.155.38:8080/v1-catalog/images/kibana/catalogIcon-kibana.svg","versionLinks":{"4.1.1-rancher1":"http://104.197.155.38:8080/v1-catalog/templates/kibana/0"},"dockerCompose":"","rancherCompose":"","questions":null,"path":"kibana"},{"type":"template","links":{"self":"http://104.197.155.38:8080/v1-catalog/templates/logspout"},"name":"Logspout","uuid":"","category":"Logging Tools","description":"Gliderlabs Logspout with Logstash adapter.\n","version":"0.2.0-rancher1","iconLink":"http://104.197.155.38:8080/v1-catalog/images/logspout/catalogIcon-gliderlabs.png","versionLinks":{"0.2.0-rancher1":"http://104.197.155.38:8080/v1-catalog/templates/logspout/0"},"dockerCompose":"","rancherCompose":"","questions":null,"path":"logspout"},{"type":"template","links":{"self":"http://104.197.155.38:8080/v1-catalog/templates/logstash"},"name":"Logstash","uuid":"","category":"ELK","description":"Logstash 1.5.3-1\nCentralize data processing of all types\n","version":"1.5.3-1-rancher1","iconLink":"http://104.197.155.38:8080/v1-catalog/images/logstash/catalogIcon-logstash.svg","versionLinks":{"1.5.3-1-rancher1":"http://104.197.155.38:8080/v1-catalog/templates/logstash/0"},"dockerCompose":"","rancherCompose":"","questions":null,"path":"logstash"}]}
```

**Comment by prachidamle** [comment]
The root cause is same as rancher/rancher#2538

PR has been submitted to fix the issue.

**Comment by sangeethah** [comment]
The latest version of rancher-catalog service is not getting released with v0.47.0 release. So moving it to next milestone.

**Comment by sangeethah** [comment]
Tested with latest build from master - dec 4 and rancher-catalog service  version - v0.4.0 .

When templates get removed from catalog , the deleted template get removed from catalog list view. It could take up to 5 mts for the changes to get propagated to the UI.
